.github/workflows/changeset-utils/index.js

```diff
@@ -0,0 +1,137 @@
+  });
+
+  return filteredCommits.map((commit) => {
+    return {
+      commit_sha: commit.sha.slice(0, 7), // first 7 characters of the commit sha is enough to identify the commit
+      commit_message: commit.commit.message,
+    };
+  });
+}
+
+const getReleasedPackages = async (pullRequest, github) => {
+  const files = await github.paginate(github.rest.pulls.listFiles.endpoint.merge({
+    owner: pullRequest.base.repo.owner.login,
+    repo: pullRequest.base.repo.name,
+    pull_number: pullRequest.number,
+  }));
+
+  const releasedPackages = [];
+  const ignoredFiles = ['README.md', 'CHANGELOG.md', './changeset/README.md', 'package.json', 'package-lock.json', 'yarn.lock', 'pnpm-lock.yaml'];
+  for (const file of files) {
+    if (!ignoredFiles.includes(file.filename)) {
+      const cwd = path.resolve(path.dirname(file.filename));
+      const pack = await readPackageUp({ cwd });
+      if (pack && pack?.packageJson?.name && !releasedPackages.includes(pack.packageJson.name)) {
+        releasedPackages.push(pack.packageJson.name);
+      }
+    } 
+  }
+
+  console.debug('Filenames', files.map((file) => file.filename));
+  return releasedPackages;
+}
+
+const getReleaseNotes = async (pullRequest, github) => {
+  const commits = await getFormattedCommits(pullRequest, github);
+  /**
+   * Release notes are generated from the commits.
+   * Format:
+   * - title
+   * - commit_sha: commit_message (Array of commits)
+   */
+  const releaseNotes = pullRequest.title + '\n\n' + commits.map((commit) => {
+    return `- ${commit.commit_sha}: ${commit.commit_message}`;
+  }).join('\n');
+
+  return releaseNotes;
+}
+
+const getChangesetContents = async (pullRequest, github) => {
+  const title = pullRequest.title;
+  const releaseType = title.split(':')[0];
+  let releaseVersion = 'patch';
+  switch (releaseType) {
+    case 'fix':
+      releaseVersion = 'patch';
+    case 'feat':
+      releaseVersion = 'minor';
+    case 'fix!':
+      releaseVersion = 'major';
+    case 'feat!':
+      releaseVersion = 'major';
+    default:
+      releaseVersion = 'patch';
+  }
+
+  const releaseNotes = await getReleaseNotes(pullRequest, github);
+  const releasedPackages = await getReleasedPackages(pullRequest, github);
+
+  if (releasedPackages.length === 0) {
+    console.debug('No packages released');
+    return '';
+  }
+  console.debug('Released packages', releasedPackages);
+  console.debug('Release notes', releaseNotes);
+
+  const changesetContents = `---\n` + releasedPackages.map((pkg) => {
+    return `'${pkg}': ${releaseVersion}`;
+  }).join('\n') + `\n---\n\n${releaseNotes}\n\n`
+
+  return changesetContents;
+};
+
+/**
+ * This function checks if a comment has already been created by the workflow.
+ * If not, it creates a comment with the changeset.
+ * If it is already created, it updates the comment with the new changeset.
+ */
+const commentWorkflow = async (pullRequest, github, changesetContents) => {
```

Comment:
just remember only maintainers can add labels. 

and maybe add some instruction that in case changeset is not needed because the pr should not trigger release, then tell them to remove changeset and modify prefix of the PR

Reply:
Done